src/main/scala/arcana/world/Main.scala

```diff
@@ -1,8 +1,19 @@
 package arcana.world
 
 object Main {
+  case class ArgumentFlags(typeInfo: Boolean)
+  def evaluatArguments(args: Array[String]): ArgumentFlags = {
+    val needTypeInfo =
+      args.foldLeft(false)((flag, arg) => flag || arg.equals("--type"))
+    ArgumentFlags(needTypeInfo)
+  }

```

Comment:
You may need to move this part into a new file. Nested classes are not necessary here.